src/app/media/media.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/publishReplay';
+import 'rxjs/add/operator/startWith';
+import { Observable } from 'rxjs/Observable';
+
+import { MatchOperator, Media, MediaBreakpoint } from './models';
+
+@Injectable()
+export class MzMediaService {
+
+  media: Observable<Media>;
+
+  // based on nodes_modules/materialize-css/sass/components/_variable.scss
+  private readonly mediaBreakpoints: MediaBreakpoint[] = [
+    { alias: 's', minWidth: 0, maxWidth: 600 },
+    { alias: 'm', minWidth: 601, maxWidth: 992 },
+    { alias: 'l', minWidth: 993, maxWidth: 1200 },
+    { alias: 'xl', minWidth: 1201, maxWidth: Number.MAX_VALUE },
+  ];
+
+  private readonly matchOperators: MatchOperator[] = [
+    {
+      operator: 'gt',
+      match: (breakpoint: MediaBreakpoint) => window.innerWidth > breakpoint.maxWidth,
+    },
+    {
+      operator: 'lt',
+      match: (breakpoint: MediaBreakpoint) => window.innerWidth < breakpoint.minWidth,
+    },
+    {
+      operator: null,
+      match: (breakpoint: MediaBreakpoint) => window.innerWidth >= breakpoint.minWidth && window.innerWidth <= breakpoint.maxWidth,
+    },
+  ];
+
+  constructor() {
+    this.media = this.registerWindowResizeListener();
+  }
+
+  isActive(breakpoint: string): Observable<boolean> {
+    return new Observable<boolean>(subscriber => {
+      this.media.subscribe((media: Media) => {
+        subscriber.next(this.isActiveBreakpoint(breakpoint));
+      });
+    });
+  }
+
+  private registerWindowResizeListener(): Observable<Media> {
+    return Observable.fromEvent(window, 'resize')
+      .map(() => this.getWindowMedia())
+      .startWith(this.getWindowMedia())
+      .publishReplay(1)
+      .refCount();
```

Comment:
It's not friendly to read the code when we do not know the Observable very well. Maybe we can add a comment that explain what the 4 lines above do.